Realm/RLMObjectBase.mm

```diff
@@ -65,7 +65,12 @@ - (instancetype)initWithObject:(id)value schema:(RLMSchema *)schema {
         array = RLMValidatedArrayForObjectSchema(array, _objectSchema, schema);
         NSArray *properties = _objectSchema.properties;
         for (NSUInteger i = 0; i < array.count; i++) {
-            [self setValue:array[i] forKeyPath:[properties[i] name]];
+            id propertyValue = array[i];
+            // strip out NSNull before passing values to standalone setters
+            if (propertyValue == NSNull.null) {
```

Comment:
`if (x == NSNull.null) { x = nil; }` appears enough places that it could use a helper function.